src/libcore/num/mod.rs

```diff
@@ -2345,7 +2345,11 @@ macro_rules! uint_impl {
         pub fn next_power_of_two(self) -> Self {
             let bits = size_of::<Self>() * 8;
             let one: Self = 1;
-            one << ((bits - self.wrapping_sub(one).leading_zeros() as usize) % bits)
+            if self == 0 {
+                1
+            } else {
+                one << (bits - self.wrapping_sub(one).leading_zeros() as usize)
```

Comment:
Technically, you can replace `self.wrapping_sub(one)` with `(self - one)`.
I think that'd make the intentions clearer (we just want to subtract one; we don't care about wrapping).